x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -281,8 +286,11 @@ export type BulkEditOperation =
       value?: undefined;
     };
 
-type RuleParamsModifier<Params extends RuleTypeParams> = (params: Params) => Promise<Params>;
-
+type RuleParamsModifier<Params extends RuleTypeParams> = (params: Params) => Promise<{
+  modifiedParams: Params;
+  isParamsUpdateSkipped: boolean;
+  skipReasons: BulkEditSkipReason[];
```

Comment:
Ended up removing `skipReason` from the `RuleParamsModifierResult`, since the params modifier no longer provides detailed reasons for why the rule was skipped.

For now, if both update of attributes and params is skipped, then the single `skip_reason` that will be returned is `RULE_NOT_MODIFIED`.